pep-0701.rst

```diff
@@ -0,0 +1,366 @@
+   that the new PEG parser, originally introduced in :pep:`617`, has allowed. The
+   improvements that these error messages brought has been greatly celebrated but
+   unfortunately f-strings cannot benefit from them because they are parsed in a
+   separate piece of the parsing machinery. This is especially unfortunate, since
+   there are several syntactical features of f-strings that can be confusing due
+   to the different implicit tokenization that happens inside the expression
+   part (for instance ``f"{y:=3}"`` is not an assignment expression).
+
+#. Other Python implementations have no way to know if they have implemented
+   f-strings correctly because contrary to other language features, they are not
+   part of the :ref:`official Python grammar <f-strings>`.
+   This is important because several prominent
+   alternative implementations are using CPython's PEG parser, `such as PyPy`_,
+   and/or are basing their grammars on the official PEG Grammar. The
+   fact that f-strings use a separate parser prevents these alternative implementations
+   from leveraging the official grammar and benefiting from improvements in error messages derived
+   from the grammar.
+
+
+A version of this proposal was originally `discussed on Python-Dev`_  and
+`presented at the Python Language Summit 2022`_ where it was enthusiastically
+received.
+
+Rationale
+=========
+
+By building on top of the new Python PEG Parser (:pep:`617`), this PEP proposes
+to redefine “f-strings”, especially emphasizing the clear separation of the
+string component and the expression (or replacement, ``{...}``) component. :pep:`498`
+summarizes the syntactical part of “f-strings” as the following:
+
+    In Python source code, an f-string is a literal string, prefixed with ‘f’, which
+    contains expressions inside braces. The expressions are replaced with their values.
+
+However unlike that definition, :pep:`498` also had formal list of exclusions on what
+can or cannot be contained inside the expression component (primarily due to the
+limitations of the existing parser). By clearly establishing the formal grammar, we
+now also have the ability to define the expression component of an f-string as truly "any
+applicable Python expression" (in that particular context) without being bound
+by the limitations imposed by the details of our implementation.
+
+The formalization effort and the premise above also has a significant benefit for
+Python programmers due to its ability to simplify and eliminate the obscure
+limitations. This reduces the mental burden and the cognitive complexity of
+f-string literals (as well as the Python language in general).
+
+#. The expression component can include any string literal that a normal Python expression
+   can include. This opens up the possibility of nesting string literals (formatted or
+   not) inside the expression component of an f-string with the same quote type (and length)::
+
+    >>> f"{"hello"}"
```

Comment:
Personally I've encountered this most often when writing something like `f"these are the things: {", ".join(things)}"`. That might be a more compelling example. (But feel free to ignore this and keep the current examples.)

Reply:
Oh, I think this is a fantastic suggestion!

@isidentical @lysnikolaou Thoughts?

Reply:
Indeed, that seems like a much real-worldy example!

Reply:
Agreed for sure!

Reply:
Made @JelleZijlstra 's comment into an applyable suggestion:

```suggestion
    >>> fTthese are the things: {", ".join(things)}"
```